components/feature/webcompat-reporter/src/main/java/mozilla/components/feature/webcompat/reporter/WebCompatReporterFeature.kt

```diff
@@ -4,25 +4,48 @@
 
 package mozilla.components.feature.webcompat.reporter
 
+import mozilla.components.concept.engine.webextension.MessageHandler
+import mozilla.components.concept.engine.webextension.Port
 import mozilla.components.concept.engine.webextension.WebExtensionRuntime
 import mozilla.components.support.base.log.logger.Logger
+import org.json.JSONObject
 
 /**
- * Feature to enable website-hotfixing via the Web Compatibility System-Addon.
+ * A feature that enables users to report site issues to Mozilla's Web Compatibility team for
+ * further diagnosis.
  */
 object WebCompatReporterFeature {
     private val logger = Logger("mozac-webcompat-reporter")
 
     internal const val WEBCOMPAT_REPORTER_EXTENSION_ID = "webcompat-reporter@mozilla.org"
     internal const val WEBCOMPAT_REPORTER_EXTENSION_URL = "resource://android/assets/extensions/webcompat-reporter/"
+    internal const val WEBCOMPAT_REPORTER_MESSAGING_ID = "mozacWebcompatReporter"
+
+    private class WebcompatReporterBackgroundMessageHandler(
+        // This information will be provided as a browser-XXX label to the reporting backend, allowing
+        // us to differentiate different android-components based products.
+        private val productName: String
+    ) : MessageHandler {
+        override fun onPortConnected(port: Port) {
+            port.postMessage(JSONObject().put("productName", productName))
+        }
+    }
 
     /**
      * Installs the web extension in the runtime through the WebExtensionRuntime install method
+     *
+     * @param runtime a WebExtensionRuntime.
+     * @param productName a custom product name used to automatically label reports. Defaults to
+     * "android-components".
      */
-    fun install(runtime: WebExtensionRuntime) {
+    fun install(runtime: WebExtensionRuntime, productName: String = "android-components") {
         runtime.installWebExtension(WEBCOMPAT_REPORTER_EXTENSION_ID, WEBCOMPAT_REPORTER_EXTENSION_URL,
             onSuccess = {
                 logger.debug("Installed WebCompat Reporter webextension: ${it.id}")
+                it.registerBackgroundMessageHandler(
```

Comment:
Would be nice to cover this with a unit test (see our other built-in extension features), basically to verify that we're sending down the product name. We can file a follow-up issue for the test, but ultimately we want to cover it. :)

Reply:
I absolutely agree. The lack of test coverage here (and, for what it's worth, for the other component, `feature-webcompat`) is worrying to me as well. The situation is a bit more complex since we try to share code between desktop and android-components (previously even Fennec), so that's a bit more annoying than I hope it to be.

I already started writing some tests both components, but I'm not yet at a point where I'd call that good, so I didn't include them here. This is still high in my priority list, and I just decided to push this PR ahead of time because this is actively blocking a coworker who is working on metric-related work. I'll absolutely continue that work, though, especially since there are more changes to this component coming soon'ish. :)

Reply:
OK. I filed a follow-up ticket: https://github.com/mozilla-mobile/android-components/issues/8012 so we don't forget. :)